frontend/js/src/settings/missing-data/MultiTrackMBIDMappingModal.tsx

```diff
@@ -0,0 +1,519 @@
+
+type MatchingTracksResult = MBTrackWithAC & {
+  searchString: string;
+};
+
+export type MatchingTracksResults = {
+  [recording_msid: string]: MatchingTracksResult;
+};
+
+export type MultiTrackMBIDMappingModalProps = {
+  releaseName: string | null;
+  missingData: Array<MissingMBData>;
+};
+
+export default NiceModal.create(
+  ({ missingData, releaseName }: MultiTrackMBIDMappingModalProps) => {
+    const modal = useModal();
+    const { APIService, currentUser } = React.useContext(GlobalAppContext);
+    const { lookupMBRelease } = APIService;
+    const { auth_token } = currentUser;
+
+    const { resolve, visible } = modal;
+
+    const [matchingTracks, setMatchingTracks] = React.useState<
+      MatchingTracksResults
+    >();
+    const [includeArtistName, setIncludeArtistName] = React.useState(false);
+    const [selectedAlbumMBID, setSelectedAlbumMBID] = React.useState<string>();
+    const [selectedAlbum, setSelectedAlbum] = React.useState<
+      MBReleaseWithMetadata
+    >();
+    const [potentialTracks, setPotentialTracks] = React.useState<
+      MBTrackWithAC[]
+    >();
+
+    const closeModal = React.useCallback(() => {
+      modal.hide();
+      document?.body?.classList?.remove("modal-open");
+      // Need to manually remove the modal backdrop
+      const backdrop = document?.body?.getElementsByClassName(
+        "modal-backdrop"
+      )?.[0];
+      backdrop?.parentElement?.removeChild(backdrop);
+      setTimeout(modal.remove, 200);
+    }, [modal]);
+
+    React.useEffect(() => {
+      const closeOnEscape = (e: KeyboardEvent) => {
+        if (e.key === "Escape") {
+          closeModal();
+        }
+      };
+      window.addEventListener("keydown", closeOnEscape);
+      return () => window.removeEventListener("keydown", closeOnEscape);
+    }, [closeModal]);
+
+    const handleError = React.useCallback(
+      (error: string | Error, title?: string): void => {
+        if (!error) {
+          return;
+        }
+        toast.error(
+          <ToastMsg
+            title={title || "Error"}
+            message={typeof error === "object" ? error.message : error}
+          />,
+          { toastId: "linked-track-error" }
+        );
+      },
+      []
+    );
+
+    const submitMBIDMapping = React.useCallback(
+      async (event: React.FormEvent) => {
+        event.preventDefault();
+
+        if (!matchingTracks || !size(matchingTracks) || !auth_token) {
+          return;
+        }
+        const promises: Promise<{ status: string }>[] = [];
+        const entries = Object.entries(matchingTracks);
+        // eslint-disable-next-line no-restricted-syntax
+        for (const [recordingMSID, trackMetadata] of entries) {
+          const recordingMBID = trackMetadata.recording.id;
+          if (recordingMBID) {
+            promises.push(
+              APIService.submitMBIDMapping(
+                auth_token,
+                recordingMSID,
+                recordingMBID
+              )
```

Comment:
Inside this callback function, we can simply pass `submitMBIDMapping` instead of passing `APIService` as a dependency.